### PR TITLE
Set up linting

### PR DESCRIPTION
**Pull request by brian386**
## Notion ticket link

[Set up linting](https://www.notion.so/uwblueprintexecs/Setup-linting-81cd22db37254845a852a28a16c57dfe)

## Implementation description
* Linting: used [this commit](https://github.com/uwblueprint/rowan-house/commit/871dee67239406a87a4bbeafd46c506eb9d73f6b) as reference

## Steps to test
1.  Linting: check if code gets formatted when you push

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        types: [file]
+        language: system
+        entry: ./scripts/lint.sh felint
+    -   id: belint
+        name: Lint backend files
+        files: ^backend/.*\.[jt]sx?$
+        types: [file]
+        language: system
+        entry: ./scripts/lint.sh belint
+
```
Comment (mahadzaryab1) [inline .pre-commit-config.yaml:16]: extra space here?

File: backend/testUtils/testDb.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ const mongo = new MongoMemoryServer();
 const mongoTest = {
   connect: async (): Promise<void> => {
     await mongo.start();
-    const uri = mongo.getUri();
+    const uri = await mongo.getUri();
```
Comment (mahadzaryab1) [inline backend/testUtils/testDb.ts]: why was this modified for linting?

**Review by mahadzaryab1 — COMMENTED** [review]
lgtm - added a couple of comments. Can you also add a screenshot of the linting here (in the console) for reference?

File: backend/testUtils/testDb.ts (hunk 2 above)
Comment (brian386) [reply backend/testUtils/testDb.ts]: oh, I'm not sure but it was giving an error if I didn't have it so I added it

**Comment by brian386** [comment]
> lgtm - added a couple of comments. Can you also add a screenshot of the linting here (in the console) for reference

How do I screenshot the linting? Should I run yarn fix and screenshot the output?

**Review by joyce-shi — CHANGES_REQUESTED** [review]
agree w/ Mahad - could we verify that this works before merging into `staging`?

**Review by carissa-tang — COMMENTED** [review]
I don't think we need the two package-lock.json files

**Comment by carissa-tang** [comment]
Did a quick test of this PR. Seems to work after doing the following...

For first time, should get this message telling you to install `pre-commit`
image

**If** you get the following error, run `chmod +x scripts/lint.sh` to make lint.sh script executable.
image

After running `docker-compose up` again, and trying to make a commit, `yarn fix` will be automatically run.
image

@brian386 Can you test this locally and see if you get the same results?

**Comment by carissa-tang** [comment]
pr should be good to go! had some issues w/ windows compatibility lol (different line ending settings - see [here](https://stackoverflow.com/questions/10418975/how-to-change-line-ending-settings) fyi)

could you @joyce-shi test locally before approving just to make sure everything is good on your end? thanks :)

**Comment by joyce-shi** [comment]
> pr should be good to go! had some issues w/ windows compatibility lol (different line ending settings - see [here](https://stackoverflow.com/questions/10418975/how-to-change-line-ending-settings) fyi)
> 
> could you @joyce-shi test locally before approving just to make sure everything is good on your end? thanks :)

LGTM! @brian386 let's merge please

**Review by joyce-shi — APPROVED** [review]
